src/Common/FitOnFhir.Common.Tests/AuthStateServiceTests.cs

```diff
@@ -94,12 +95,41 @@ public void GivenAnonymousLoginEnabledWithValidRequest_WhenCreateAuthStateCalled
             }
         }
 
+        [InlineData(false, true, true, "external_id")]
+        [InlineData(false, true, false, "external_id, redirect_url")]
+        [InlineData(false, false, true, "external_id, external_system")]
+        [InlineData(false, false, false, "external_id, external_system, redirect_url")]
+        [InlineData(true, true, false, "redirect_url")]
+        [InlineData(true, false, true, "external_system")]
+        [InlineData(true, false, false, "external_system, redirect_url")]
+        [Theory]
+        public void GivenAnonymousLoginEnabledWithInvalidRequest_WhenCreateAuthStateCalled_ThrowsAuthStateExceptionWithCorrectMessage(
+            bool includePatient,
+            bool includeSystem,
+            bool includeRedirectUrl,
+            string missing)
+        {
+            SetupConfiguration(true);
+            SetupHttpRequest(ExpectedToken, includePatient: includePatient, includeSystem: includeSystem, includeRedirectUrl: includeRedirectUrl);
+
+            try
+            {
+                var authState = _authStateService.CreateAuthState(Request);
```

Comment:
In all the tests that use the try/catch, you will need to add an assert after you make the request to ensure that the test doesn't pass because no exception is thrown or a different exception is thrown.

```c#
try
{
    var authState = _authStateService.CreateAuthState(Request);
    Assert.True(false, "Expected exception not thrown."); // CreateAuthState() did not throw an exception
}
catch (AuthStateException ex)
```